.github/workflows/test_jest_performance_cronjob.yml

```diff
@@ -0,0 +1,139 @@
+          # Set different duplicate numbers based on trigger type
+          if [ "${{ github.event_name }}" = "schedule" ]; then
+            echo "DUPLICATE_NUMBER_BASE=15" >> $GITHUB_ENV
+          else
+            echo "DUPLICATE_NUMBER_BASE=5" >> $GITHUB_ENV
+          fi
+
+      - name: Run Services with Docker
+        run: |
+          npm run start:services:ci
+
+      - name: Run performance tests with Jest
+        working-directory: ${{ env.workingDirectory }}
+        run: |
+          # Run different tests with different duplicate numbers
+          DUPLICATE_NUMBER=15 docker compose exec 121-service npm run test:integration:all -t BulkUpdate32kRegistrations.test.ts --shard=${{ matrix.shard }}/${{ matrix.shardCount }}
+          DUPLICATE_NUMBER=17 docker compose exec 121-service npm run test:integration:all -t FindDuplicates100kRegistrations.test.ts --shard=${{ matrix.shard }}/${{ matrix.shardCount }}
+          DUPLICATE_NUMBER=5 docker compose exec 121-service npm run test:integration:all -t GetProgramWithManyAttributes.test.ts --shard=${{ matrix.shard }}/${{ matrix.shardCount }}
+          DUPLICATE_NUMBER=17 docker compose exec 121-service npm run test:integration:all -t Payment100kRegistrationIntersolveVisa.test.ts --shard=${{ matrix.shard }}/${{ matrix.shardCount }}
+          DUPLICATE_NUMBER=17 docker compose exec 121-service npm run test:integration:all -t Payment100kRegistrationSafaricom.test.ts --shard=${{ matrix.shard }}/${{ matrix.shardCount }}
+          DUPLICATE_NUMBER=15 docker compose exec 121-service npm run test:integration:all -t PerformanceDuringPayment.test.ts --shard=${{ matrix.shard }}/${{ matrix.shardCount }}
+          DUPLICATE_NUMBER=15 docker compose exec 121-service npm run test:integration:all -t StatusChangePaymentInLargeProgram.test.ts --shard=${{ matrix.shard }}/${{ matrix.shardCount }}
+          docker compose exec 121-service npm run test:integration:all -t Import1000Registrations.test.ts --shard=${{ matrix.shard }}/${{ matrix.shardCount }}
+
+      # For more info, see TODO: README.md
+      - name: Generate lcov report from 121-service
+        working-directory: ${{ env.workingDirectory }}
+        run: |
+          curl -d '{"secret":"fill_in_secret"}' -H "Content-Type: application/json" -X POST '${{ env.EXTERNAL_121_SERVICE_URL }}/api/test/kill-service' || true
+          docker compose exec 121-service npm run coverage:report:integration
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: integration-test-coverage-report-${{ matrix.shard }}
+          path: |
+            ${{ env.workingDirectory }}coverage
+
+      - name: Docker logs
+        if: always()
+        uses: jwalton/gh-docker-logs@v2
+
+  # This job is here to make the above jobs "required" in GitHub.
+  # This is a workaround for this issue: https://github.com/orgs/community/discussions/44490
+  test-shard-resolution-api:
```

Comment:
the old k6 test also didn't need this setup. You've seemed to copy a lot here from the test_service_api.yml, which makes sense for some parts but maybe not for all..